public/pages/DetectorsList/List/List.tsx

```diff
@@ -276,13 +306,131 @@ export const DetectorList = (props: ListProps) => {
     }));
   };
 
+  const handleSelectionChange = (currentSelected: DetectorListItem[]) => {
+    setSelectedDetectorsForAction(currentSelected);
+  };
+
+  const handleStartDetectorsAction = () => {
+    const validDetectors = getValidDetectors(
+      selectedDetectorsForAction,
+      DETECTOR_ACTION.START
+    );
+    if (!isEmpty(validDetectors)) {
+      const confirmStartDetectorsModal = (
+        <ConfirmStartDetectorsModal
+          detectors={validDetectors}
+          hideModal={hideConfirmModal}
+          onStartDetectors={handleStartDetectorJobs}
+        />
+      );
+      setShowConfirmModal(true);
+      setConfirmModal(confirmStartDetectorsModal);
+    } else {
+      toastNotifications.addDanger(
+        'All selected detectors are unable to start. Make sure selected \
+          detectors have features and are not already running'
+      );
+    }
+  };
+
+  const handleStopDetectorsAction = () => {
+    const validDetectors = getValidDetectors(
+      selectedDetectorsForAction,
+      DETECTOR_ACTION.STOP
+    );
+    if (!isEmpty(validDetectors)) {
+      const confirmStopDetectorsModal = (
+        <ConfirmStopDetectorsModal
+          detectors={validDetectors}
+          monitors={allMonitors}
+          hideModal={hideConfirmModal}
+          onStopDetectors={handleStopDetectorJobs}
+        />
+      );
+      setShowConfirmModal(true);
+      setConfirmModal(confirmStopDetectorsModal);
+    } else {
+      toastNotifications.addDanger(
+        'All selected detectors are unable to stop. Make sure selected \
+          detectors are already running'
+      );
+    }
+  };
+
+  const handleDeleteDetectorsAction = async () => {
+    // stub for now, will add implementation in later PR
+  };
+
+  const handleStartDetectorJobs = async () => {
+    const validIds = getValidDetectors(
+      selectedDetectorsForAction,
+      DETECTOR_ACTION.START
+    ).map(detector => detector.id);
+    const promises = validIds.map(async (id: string) => {
+      return dispatch(startDetector(id));
+    });
+    await Promise.all(promises)
+      .then(() => {
+        toastNotifications.addSuccess(
+          'All selected detectors have been started successfully'
+        );
+      })
+      .catch(error => {
+        toastNotifications.addDanger(
+          `Error starting all selected detectors: ${error}`
```

Comment:
If multiple detectors fail, will show multiple toast notifications? Can we aggregate all errors and show one toast notification?

Reply:
`Promise.all()` tries to resolve all of the promises. If any fails, it aborts everything and throws an exception. So this will only show once.

Reply:
The same thing applies to the success toast notification; it will only be shown once after a batch action is performed.

Reply:
will it only show error for 1st failed detector, even there are multiple failed detectors?

Reply:
> will it only show error for 1st failed detector, even there are multiple failed detectors?

Kind of. Because `Promise.all()` fails out once a single failure has occurred, the rest of the promises won't be resolved, so technically only one detector could fail. But yes, the error message here would be related to the first failed detector